### PR TITLE
feat(builders): add new command permissions v2

### PR DESCRIPTION
**Pull request by vladfrangu**
**Please describe the changes this PR makes and why it should be merged:**

Adds support for command permissions v2 in builders

**Status and versioning classification:**

- Code changes have been tested against the Discord API, or there are no code changes
- I know how to update typings and have done so, or typings don't need updating
- This PR changes the library's interface (methods or parameters added)
- This PR includes breaking changes (methods removed or renamed, parameters moved or removed)

### CODE REVIEW

**Review by SuperchupuDev — COMMENTED** [review]
Shouldn't it be `setDMPermission` instead of `setDmPermission`? It's used like that pretty much everywhere in discord.js

**Comment by vladfrangu** [comment]
> Shouldn't it be `setDMPermission` instead of `setDmPermission`? It's used like that pretty much everywhere in discord.js

🤷 . Leaving it for the core maintainers to decide, I think its fine as is

**Comment by kyranet** [comment]
we should switch to `setDMPermissions`. While I dislike having 3 continuous uppercase letters, it's not unseen in JavaScript (`URLSearchParams` for example).

**Comment by vladfrangu** [comment]
If the other two maintainers agree, I'll do the change :D

File: packages/builders/src/interactions/contextMenuCommands/Assertions.ts [hunk 1]
```diff
@@ -30,3 +30,19 @@ export function validateRequiredParameters(name: string, type: number) {
...
+	dmPermissionPredicate.parse(value);
+}
+
+const memberPermissionPredicate = s.union(
+	s.bigint.transform((value) => value.toString()),
+	s.number.safeInt.transform((value) => value.toString()),
+	s.string.regex(/^\d+$/),
+).nullish;
+
+export function validateDefaultMemberPermission(permissions: unknown) {
```
Comment (xHyroM) [inline packages/builders/src/interactions/contextMenuCommands/Assertions.ts:46]:
```suggestion
export function validateDefaultMemberPermissions(permissions: unknown) {
```

**Review by xHyroM — CHANGES_REQUESTED** [review]

File: packages/builders/src/interactions/contextMenuCommands/ContextMenuCommandBuilder.ts [hunk 2]
```diff
@@ -15,9 +26,21 @@ export class ContextMenuCommandBuilder {
 	/**
 	 * Whether the command is enabled by default when the app is added to a guild
 	 *
-	 * @default true
+	 * @deprecated This property is deprecated and will be removed in the future.
+	 * You should use `setDefaultMemberPermissions` or `setDmEnabled` instead.
```
Comment (SuperchupuDev) [inline packages/builders/src/interactions/contextMenuCommands/ContextMenuCommandBuilder.ts:30]:
```suggestion
	 * You should use `setDefaultMemberPermissions` or `setDMPermission` instead.
```

File: packages/builders/src/interactions/slashCommands/SlashCommandBuilder.ts [hunk 3]
```diff
@@ -44,9 +46,21 @@ export class SlashCommandBuilder {
 	/**
 	 * Whether the command is enabled by default when the app is added to a guild
 	 *
-	 * @default true
+	 * @deprecated This property is deprecated and will be removed in the future.
+	 * You should use `setDefaultMemberPermissions` or `setDmEnabled` instead.
```
Comment (SuperchupuDev) [inline packages/builders/src/interactions/slashCommands/SlashCommandBuilder.ts:50]:
```suggestion
	 * You should use `setDefaultMemberPermissions` or `setDMPermission` instead.
```

File: packages/builders/src/interactions/slashCommands/SlashCommandBuilder.ts [hunk 4]
```diff
@@ -74,12 +84,48 @@ export class SlashCommandBuilder {
 	 * @param value Whether or not to enable this command by default
 	 *
 	 * @see https://discord.com/developers/docs/interactions/application-commands#permissions
+	 * @deprecated Use `setDefaultMemberPermissions` and `setDMPermission` instead.
```
Comment (SuperchupuDev) [inline packages/builders/src/interactions/slashCommands/SlashCommandBuilder.ts:87]:
```suggestion
	 * @deprecated Use `setDefaultMemberPermissions` or `setDMPermission` instead.
```
consistency with the other `@deprecated` tags

File: packages/builders/src/interactions/contextMenuCommands/Assertions.ts [hunk 5]
```diff
@@ -30,3 +30,19 @@ export function validateRequiredParameters(name: string, type: number) {
 	// Assert type is valid
 	validateType(type);
 }
+
+const dmPermissionPredicate = s.boolean.nullish;
+
+export function validateDmPermission(value: unknown): asserts value is boolean | null | undefined {
```
Comment (SuperchupuDev) [inline packages/builders/src/interactions/contextMenuCommands/Assertions.ts:36]: Shouldn't this be `validateDMPermission`?

**Review by SpaceEEC — APPROVED** [review]

**Review by iCrawl — APPROVED** [review]

**Comment by iCrawl** [comment]
This needs a rebase.

**Comment by pierluigizagaria** [comment]
Please approve this merge!

**Pull request merged**